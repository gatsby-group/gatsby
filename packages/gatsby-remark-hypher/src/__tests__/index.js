const Remark = require(`remark`)
const visit = require(`unist-util-visit`)

const plugin = require(`../`)

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

describe(`gatsby-remark-hypher`, () => {
  it(`applies hyphenation to text nodes`, () => {
    const markdownAST = remark.parse(`This is a ridiculously long word.`)

    const transformed = plugin({ markdownAST })

    visit(transformed, `text`, node => {
      // transformed text node should contain a soft hyphen
      expect(node.value).toEqual(expect.stringContaining(`­`))
      expect(node.value).toMatchSnapshot()
    })
  })
  it(`does not apply hyphenation to non text nodes`, () => {
    const markdownAST = remark.parse(`This is a \`ridiculously\` long word.`)

    const transformed = plugin({ markdownAST })

    visit(transformed, `text`, node => {
      // transformed text nodes should not contain any soft hyphens
      expect(node.value).toEqual(expect.not.stringContaining(`­`))
      expect(node.value).toMatchSnapshot()
    })
  })
})
