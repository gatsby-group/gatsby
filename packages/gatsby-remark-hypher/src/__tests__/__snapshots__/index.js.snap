// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-hypher applies hyphenation to text nodes 1`] = `"This is a ridicuÂ­lously long word."`;

exports[`gatsby-remark-hypher does not apply hyphenation to non text nodes 1`] = `"This is a "`;

exports[`gatsby-remark-hypher does not apply hyphenation to non text nodes 2`] = `" long word."`;
